crates/rust-analyzer/src/caps.rs

```diff
@@ -54,7 +54,7 @@ pub fn server_capabilities(client_caps: &ClientCapabilities) -> ServerCapabiliti
         code_action_provider: Some(code_action_capabilities(client_caps)),
         code_lens_provider: Some(CodeLensOptions { resolve_provider: Some(true) }),
         document_formatting_provider: Some(OneOf::Left(true)),
-        document_range_formatting_provider: None,
+        document_range_formatting_provider: Some(OneOf::Left(true)),
```

Comment:
Should we perhaps do somthing like `config.enable_range_formatting.then(|| OneOf::Left(true))` ?

Reply:
I think it could go either way. One thing I like about the way it is currently is that if a user tries to use without opting in, the error message will tell them to opt-in. Otherwise it would just say that the method is unsupported.